doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,653 @@
+        """
+        pass
+
+**Bad:**
+
+.. code-block:: python
+
+    def astype(dtype):
+        """Casts Series type.
+
+        Verb in third-person of the present simple, should be infinitive.
+        """
+        pass
+
+    def astype(dtype):
+        """Method to cast Series type.
+
+        Does not start with verb.
+        """
+        pass
+
+    def astype(dtype):
+        """Cast Series type
+
+        Missing dot at the end.
+        """
+        pass
+
+    def astype(dtype):
+        """Cast Series type from its current type to the new type defined in
+        the parameter dtype.
+
+        Summary is too verbose and doesn't fit in a single line.
+        """
+        pass
+
+Section 2: Extended summary
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The extended summary provides details on what the function does. It should not
+go into the details of the parameters, or discuss implementation notes, which
+go in other sections.
+
+A blank line is left between the short summary and the extended summary. And
+every paragraph in the extended summary is finished by a dot.
+
+.. code-block:: python
+
+    def unstack():
+        """Pivot a row index to columns.
+
+        When using a multi-index, a level can be pivoted so each value in
+        the index becomes a column. This is especially useful when a subindex
+        is repeated for the main index, and data is easier to visualize as a
+        pivot table.
+
+        The index level will be automatically removed from the index when added
+        as columns.
+        """
+        pass
+
+Section 3: Parameters
+~~~~~~~~~~~~~~~~~~~~~
+
+The details of the parameters will be added in this section. This section has
+the title "Parameters", followed by a line with a hyphen under each letter of
+the word "Parameters". A blank line is left before the section title, but not
+after, and not between the line with the word "Parameters" and the one with
+the hyphens.
+
+After the title, each parameter in the signature must be documented, including
+`*args` and `**kwargs`, but not `self`.
+
+The parameters are defined by their name, followed by a space, a colon, another
+space, and the type (or types). Note that the space between the name and the
+colon is important. Types are not defined for `*args` and `**kwargs`, but must
+be defined for all other parameters. After the parameter definition, it is 
+required to have a line with the parameter description, which is indented, and
+can have multiple lines. The description must start with a capital letter, and
+finish with a dot.
+
+Keyword arguments with a default value, the default will be listed in brackets
+at the end of the description (before the dot). The exact form of the
+description in this case would be "Description of the arg (default is X).". In
+some cases it may be useful to explain what the default argument means, which
+can be added after a comma "Description of the arg (default is -1, which means
+all cpus).".
```

Comment:
See also eg https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.take.html where it is done consistently

Reply:
Having the default after the description (and not the type), and having the default first if it's a set like {0, 10, 25}, is in the numpy docstring convention. I agree with you in both cases, it can be clearer having the default after the type, and the options in a consistent order. Just pointing out where these come from.

Reply:
I also like the current way, and our users might be used to it (I certainly am)